aexpect/ops.py

```diff
@@ -0,0 +1,439 @@
+    :type session: ShellSession
+    :param str path: absolute or relative path to list
+    :param bool is_glob: whether to expand or not the path as a glob pattern
+    :returns: names of files in dir (without path components)
+    :rtype: [str]
+
+    Just like :py:func:`os.listdir`, does not include file names starting with
+    dot (`'.'`)
+    """
+    cmd = f'ls {flags} {path}' if is_glob else f'ls {flags} {quote(path)}'
+    status, output = session.cmd_status_output(cmd)
+    if output:
+        output = output.strip()
+    if status == 2:      # probably just nothing found
+        return []
+    if status != 0:
+        raise RuntimeError(f'Failed to ls {path}: {output}')
+    return output.splitlines()
+
+
+def cat(session, filename):
+    """
+    Get contents of a text file from a session.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str filename: full path of file
+    :returns: file contents
+    :rtype: str
+    :raises: :py:class:`RuntimeError` if cat command fails
+
+    Should only be used for very short files without tabs or other fancy
+    contents. Otherwise better download file or use some other method.
+    """
+    cmd = 'cat ' + quote(filename)
+    status, output = session.cmd_status_output(cmd)
+    if output:
+        output = output.strip()
+    if status != 0:
+        raise RuntimeError(f'Failed to cat {filename}: {output}')
+    return output
+
+
+def make_tempdir(session, template="tmp.XXXXXXXXXX"):
+    """
+    Create a temporary directory in a safe way through a session.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str template: directory name template as used in the standard command
+    :returns: name of temporary directory
+    :rtype: str
+    :raises: :py:class:`RuntimeError` if mktemp returns non-zero exit status
+
+    Calls `mktemp -d`, refer to `man` for more info.
+    """
+    cmd = f'mktemp -d {template}'
+    status, output = session.cmd_status_output(cmd)
+    if output:
+        output = output.strip()
+    if status != 0:
+        raise RuntimeError(f'Failed to create temporary directory: {output}')
+    return output
+
+
+def make_tempfile(session, template="tmp.XXXXXXXXXX"):
+    """
+    Create a temporary file in a safe way through a session.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str template: file name template as used in the standard command
+    :returns: name of temporary file
+    :rtype: str
+    :raises: :py:class:`RuntimeError` if mktemp returns non-zero exit status
+
+    Calls `mktemp`, refer to `man` for more info.
+    """
+    cmd = f'mktemp {template}'
+    status, output = session.cmd_status_output(cmd)
+    if output:
+        output = output.strip()
+    if status != 0:
+        raise RuntimeError(f'Failed to create temporary file: {output}')
+    return output
+
+
+def move(session, source, target):
```

Comment:
The same here (as in copy, the `flags` argument) especially for `-f`.

Reply:
Nice, I added it here too before reading the rest of your comments.